frontend/src/lib/k8s/networkpolicy.tsx

```diff
@@ -0,0 +1,41 @@
+import { apiFactoryWithNamespace } from './apiProxy';
+import { KubeObjectInterface, LabelSelector, makeKubeObject } from './cluster';
+
+interface NetworkPolicyPort {
+  port: string | number;
+  protocol: string;
+}
+
+interface IPBlock {
+  cidr: string;
+  except: string;
+}
+
+interface NetworkPolicyPeer {
+  ipBlock: IPBlock;
+  namespaceSelector: LabelSelector;
+  podSelector: LabelSelector;
+}
+
+interface NetworkPolicyEgressRule {
+  ports: NetworkPolicyPort[];
+  to: NetworkPolicyPeer[];
+}
+
+interface NetworkPolicyIngressRule {
+  ports: NetworkPolicyPort[];
+  from: NetworkPolicyPeer[];
+}
+
+export interface KubeNetworkPolicy extends KubeObjectInterface {
+  egress: NetworkPolicyEgressRule;
+  ingress: NetworkPolicyIngressRule;
```

Comment:
I think these two need to be arrays (of the types you defined), as per https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#networkpolicyspec-v1-networking-k8s-io